### PR TITLE
Build migration tools in CI

### LINKED ISSUE #15037: Berkeley migration deployment with new dune profile
With new fixes do replayer, we need to adjust deployment code. 

Replayer for migration is using different dune profile. Currently Ci is able to use dune profile different than devnet, but it will also build another dockers like deamon / test executive etc. We don't want that.

### PR DESCRIPTION
**Pull request by dkijania**
Introduced new dune profile: `berkeley_archive_migration` which forced me to do a small refactoring in a way we build and publish artifacts:

- I joined build_archive_release.sh and rebuild_deb.sh since they are sharing the same code in 95%
- Refactored MinaArtifact.dhall job so we can control what dockers are built. In may cases we don't need test executive or batch txn docker. In our case we just need to build archive docker.
- Introduced Artifacts.dhall enum which defined every possible docker we can build
- Removed dependency to makefile in build artifacts as we would like to pass arguments to rebuild-deb.sh script and control which debian will be built

Resolves #15037

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
We need to wait until replayer fixes are merged before mergin this

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Comment by dkijania** [comment]
!ci-build-me

File: scripts/rebuild-deb.sh
Comment (georgeee) [inline scripts/rebuild-deb.sh:58]:
```suggestion
    # use dune profile as suffix but replace underscore to dashes so deb builder won't complain
```

Comment (georgeee) [reply scripts/rebuild-deb.sh:58]: Also I suggest to capture `${DUNE_PROFILE//_/-}` into a new variable and reuse below.

File: scripts/rebuild-deb.sh
Comment (georgeee) [inline scripts/rebuild-deb.sh:389]: Indentation here looks a bit weird (minor)

**Review by georgeee — APPROVED** [review]

**Comment by dkijania** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Comment by dkijania** [comment]
!ci-buidl-me

**Review by amc-ie — APPROVED** [review]
Approved!

**Comment by mrmr1993** [comment]
!ci-nightly-me

**Pull request merged**
